### PR TITLE
Remove attrs usage in Context

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
## Description
attrs adds overhead which we can do without.

```
scripts/perf-run-scenario tracer Datadog/dd-trace-py@master Kyle-Verhoog/dd-trace-py@non-attrs-ctx ./artifacts/
```

| Benchmark      | master | non-attrs-ctx |
|----------------|------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------|
| tracer-small   | 193 us                                                                                         | 179 us: 1.08x faster                                                                            |
| tracer-medium  | 1.73 ms                                                                                        | 1.68 ms: 1.03x faster                                                                           |
| tracer-large   | 17.7 ms                                                                                        | 16.8 ms: 1.05x faster                                                                           |
| Geometric mean | (ref)                                                                                          | 1.06x faster                                                                                    |

```
scripts/perf-run-scenario span Datadog/dd-trace-py@master Kyle-Verhoog/dd-trace-py@non-attrs-ctx ./artifacts/
```
| Benchmark      | master | non-attrs-ctx |
|----------------|----------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------|
| span-add-tags  | 143 ms                                                                                       | 140 ms: 1.02x faster                                                                          |
| Geometric mean | (ref)                                                                                        | 1.02x faster                                                                                  |

## Checklist
- [x] Added to the correct milestone.

### CODE REVIEW

**Review by P403n1x87 — APPROVED** [review]
👍   Are there any other `attrs` classes on hot paths that could benefit from a similar change?

File: ddtrace/context.py [hunk 1]
```diff
@@ -20,27 +18,33 @@
 log = get_logger(__name__)
 
 
-@attr.s(eq=False, slots=True)
```
Comment (brettlangdon) [inline ddtrace/context.py:23]: should we add in a `__str__` and a `__repr__` ?

Comment (Kyle-Verhoog) [reply ddtrace/context.py:23]: Good catch, yeah we should add these back in

File: ddtrace/context.py [hunk 2]
```diff
@@ -109,3 +110,28 @@ def clone(self):
...
+        return "Context(trace_id=%s, span_id=%s, _meta=%s, _metrics=%s)" % (
+            self.trace_id,
+            self.span_id,
+            self._meta,
+            self._metrics,
+        )
+
+    def __str__(self):
+        # type: () -> str
+        return repr(self)
```
Comment (P403n1x87) [inline ddtrace/context.py:137]:
```suggestion
    def __str__(self):
        # type: () -> str
        return repr(self)
```
__str__ = __repr__

**Comment by Kyle-Verhoog** [comment]
```python
In [2]: @attr.s()
   ...: class A(object):
   ...:     pass

In [3]: class B(object):
   ...:     pass


In [4]: %timeit A()
147 ns ± 2.3 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit B()
75.4 ns ± 0.976 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

**Review by P403n1x87 — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**